Guides for migrating data (#3865)
influxdata/docs-v2: content/influxdb/v2.1/write-data/migrate-cloud-to-oss.md

## Troubleshoot migration task failures
If the migration task fails, [view your task logs](/influxdb/v2.1/process-data/manage-tasks/task-run-history/)
to identify the specific error. Below are common causes of migration task failures.

- [Exceeded rate limits](#exceeded-rate-limits)
- [Invalid API token](#invalid-api-token)
- [Query timeout](#query-timeout)

### Exceeded rate limits
If your data migration causes you to exceed your InfluxDB Cloud organization's
limits and quotas, the task will return an error similar to:

```
too many requests
```

**Possible solutions**:
- Update the `migration.batchInterval` setting in your migration task to use
  a smaller interval. Each batch will then query less data.

### Invalid API token
If the API token you add as the `INFLUXDB_CLOUD_SECRET` doesn't have read access to
your InfluxDB Cloud bucket, the task will return an error similar to:

```
unauthorized access
```

**Possible solutions**:
- Ensure the API token has read access to your InfluxDB Cloud bucket.
- Generate a new InfluxDB Cloud API token with read access to the bucket you

>   want to migration. Then update the `INFLUXDB_CLOUD_TOKEN` secret in your

Reviewer: ```suggestion
  want to migrate. Then, update the `INFLUXDB_CLOUD_TOKEN` secret in your
```

  InfluxDB OSS instance with the new token.

### Query timeout
The InfluxDB Cloud query timeout is 90 seconds. If it takes longer than this to

> return the data from the batche's time range, the query will time out and the

Reviewer: ```suggestion
return the data from the batch interval, the query will time out and the
```

      _See [Determine your batch interval](#determine-your-batch-interval)._
    - **batchBucket**: InfluxDB OSS bucket to store migration batch metadata in.
    - **cloudHost**: [InfluxDB Cloud region URL](/influxdb/cloud/reference/regions)
      to migrate data from.
    - **cloudOrg**: InfluxDB Cloud organization to migrated data from.
    - **cloudToken**: InfluxDB Cloud API token. To keep the API token secure, store
      it as a secret in InfluxDB OSS.
    - **cloudBucket**: InfluxDB Cloud bucket to migrate data from.
    - **ossBucket**: InfluxDB OSS bucket to migrate data to.

### Migration Flux script

```js
import "array"
import "experimental"
import "influxdata/influxdb/secrets"

// Configure the task
option task = {every: 5m, name: "Migrate data from InfluxDB Cloud"}

// Configure the migration
migration = {
    start: 2022-01-01T00:00:00Z,
    stop: 2022-02-01T00:00:00Z,
    batchInterval: 1h,
    batchBucket: "migration",
    cloudHost: "https://cloud2.influxdata.com",
    cloudOrg: "example-cloud-org",
    cloudToken: secrets.get(key: "INFLUXDB_CLOUD_TOKEN"),
    cloudBucket: "example-cloud-bucket",
    ossBucket: "example-oss-bucket",
}

> // batchRange dynamically returns a record with start and stop properties for

Reviewer: 👍

{{% /expand %}}
<!------------------------ END Determine task interval ------------------------>

<!---------------------- BEGIN Determine migration start ---------------------->
{{% expand "Determine your migration start time" %}}

The `migration.start` time should be at or near the same time as the earliest
data point you want to migrate.
All migration batches are determined using the `migration.start` time and
`migration.batchInterval` settings.

To find time of the earliest point in your bucket, run the following query:

```js
from(bucket: "example-cloud-bucket")
    |> range(start: 0)
    |> group()
    |> first()
    |> keep(columns: ["_time"])
```

{{% /expand %}}
<!----------------------- END Determine migration start ----------------------->

<!----------------------- BEGIN Determine batch interval ---------------------->
{{% expand "Determine your batch interval" %}}

The `migration.batchInterval` setting controls the time range queried by each batch.
The "density" of the data in your InfluxDB Cloud bucket and your InfluxDB Cloud
organization's [rate limits and quotas](/influxdb/cloud/account-management/limits/)
determine what your batch interval should be.

> For example, if you are migrating data collected from hundreds of sensors with

Reviewer: ```suggestion
For example, if you're migrating data collected from hundreds of sensors with
```
